lib/browser.js

```diff
@@ -142,24 +142,24 @@ class Browser {
     this.clearNoActivityTimeout()
   }
 
-  onDisconnect (disconnectedSocket) {
-    this.activeSockets.splice(this.activeSockets.indexOf(disconnectedSocket), 1)
+  onDisconnect (reason, disconnectedSocket) {
+    helper.arrayRemove(this.activeSockets, disconnectedSocket)
 
     if (this.activeSockets.length) {
       this.log.debug('Disconnected %s, still have %s', disconnectedSocket.id, this.getActiveSocketsIds())
       return
     }
 
     if (this.state === CONNECTED) {
-      this.disconnect('client disconnected from CONNECTED state')
+      this.disconnect(`client disconnected from CONNECTED state (${reason})`)
```

Comment:
@johnjbarton 
Please take a look if it's good and enough way of passing reason into disconnect.